pkg/odo/cli/url/describe.go

```diff
@@ -0,0 +1,112 @@
+package url
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/machineoutput"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/odo/util"
+	"github.com/openshift/odo/pkg/odo/util/completion"
+	"github.com/openshift/odo/pkg/url"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/util/templates"
+)
+
+// DescribeRecommendedCommandName is the recommended describe command name
+const describeRecommendedCommandName = "describe"
+
+var describeExample = ktemplates.Examples(`  # Describe a URL
+%[1]s myurl
+`)
+
+// URLListOptions encapsulates the options for the odo url list command
+type URLDescribeOptions struct {
+	localConfigInfo  *config.LocalConfigInfo
+	componentContext string
+	url string
+	*genericclioptions.Context
+}
+
+// NewURLDescribeOptions creates a new URLCreateOptions instance
+func NewURLDescribeOptions() *URLDescribeOptions {
+	return &URLDescribeOptions{&config.LocalConfigInfo{}, "", "", &genericclioptions.Context{}}
+}
+
+// Complete completes URLDescribeOptions after they've been Listed
+func (o *URLDescribeOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.Context = genericclioptions.NewContext(cmd)
+	o.localConfigInfo, err = config.NewLocalConfigInfo(o.componentContext)
+	if err != nil {
+		return errors.Wrap(err, "failed intiating local config")
+	}
+	if len(args) == 1 {
+		o.url = args[0]
+	}
+	return
+}
+
+// Validate validates the URLDescribeOptions based on completed values
+func (o *URLDescribeOptions) Validate() (err error) {
+	return util.CheckOutputFlag(o.OutputFlag)
+}
+
+// Run contains the logic for the odo url list command
+func (o *URLDescribeOptions) Run() (err error) {
+
+	urls, err := url.List(o.Client, o.localConfigInfo, o.Component(), o.Application)
+	if err != nil {
+		return err
+	}
+	u, err := urls.Get(o.url)
+	if err != nil {
+		return err
+	}
```

Comment:
Since `url describe` can take only one URL name as a parameter input, I don't think we need url.List() here. We have faced slowness because of listing when we didn't need it. Please write a generic Get function for URL and use the client to make a get call for a route.

Reply:
got it , will double check how to do this

Reply:
@mik-dass 
um.. after double check , seems we have something like 
clusterURL := getMachineReadableFormat(r)
which means we can't search in etcd directly, we need get back the data then do some transform and 
this of course need list ,

I can change the function to Get() the compare in the new function but still need list
so don't know how much improvement we get can... so, still a O(n) search , not O(1)

Reply:
No you don't need a list. First write getRoute() function in occlient.go like https://github.com/openshift/odo/blob/master/pkg/occlient/occlient.go#L484 which returns the found route. Then write a Get() (don't use the existing one, it requires a list) function, call the getRoute() in occlient.go and transform the returned route to machine readable format using getMachineReadableFormat().

Reply:
ok, get your point, I will try the way you suggested, thanks!